diff -urN qt-creator-2.6.0-src/src/libs/ssh/ssh.pro qt-creator-2.6.0-src-patched/src/libs/ssh/ssh.pro
--- qt-creator-2.6.0-src/src/libs/ssh/ssh.pro	2012-11-07 00:01:49.000000000 +1100
+++ qt-creator-2.6.0-src-patched/src/libs/ssh/ssh.pro	2012-11-09 14:02:32.372799698 +1100
@@ -67,3 +67,5 @@
 FORMS = $$PWD/sshkeycreationdialog.ui
 
 include(../3rdparty/botan/botan.pri)
+
+unix: LIBS += -ldl
diff -urN qt-creator-2.6.0-src/src/libs/utils/utils.pro qt-creator-2.6.0-src-patched/src/libs/utils/utils.pro
--- qt-creator-2.6.0-src/src/libs/utils/utils.pro	2012-11-07 00:01:49.000000000 +1100
+++ qt-creator-2.6.0-src-patched/src/libs/utils/utils.pro	2012-11-09 13:57:36.573282858 +1100
@@ -18,6 +18,8 @@
 SOURCES += \
     proxyaction.cpp
 
+unix: LIBS += -lX11
+
 win32: LIBS += -luser32
 # PortsGatherer
 win32: LIBS += -liphlpapi -lws2_32
