diff -urN qt-creator-opensource-src-3.2.0/src/libs/ssh/ssh.pro qt-creator-opensource-src-3.2.0-patched/src/libs/ssh/ssh.pro
--- qt-creator-opensource-src-3.2.0/src/libs/ssh/ssh.pro	2014-08-18 18:52:02.000000000 +1100
+++ qt-creator-opensource-src-3.2.0-patched/src/libs/ssh/ssh.pro	2014-08-21 14:59:52.066946597 +1100
@@ -69,3 +69,5 @@
 FORMS = $$PWD/sshkeycreationdialog.ui
 
 include(../3rdparty/botan/botan.pri)
+
+unix: LIBS += -ldl
diff -urN qt-creator-opensource-src-3.2.0/src/libs/utils/utils.pro qt-creator-opensource-src-3.2.0-patched/src/libs/utils/utils.pro
--- qt-creator-opensource-src-3.2.0/src/libs/utils/utils.pro	2014-08-18 18:52:02.000000000 +1100
+++ qt-creator-opensource-src-3.2.0-patched/src/libs/utils/utils.pro	2014-08-21 14:59:52.066946597 +1100
@@ -21,3 +21,4 @@
 win32: LIBS += -luser32 -lshell32
 # PortsGatherer
 win32: LIBS += -liphlpapi -lws2_32
+unix: LIBS += -lX11
