diff -up qt-creator-opensource-src-19.0.0/src/libs/3rdparty/yaml-cpp/src/emitterutils.cpp.omv~ qt-creator-opensource-src-19.0.0/src/libs/3rdparty/yaml-cpp/src/emitterutils.cpp
--- qt-creator-opensource-src-19.0.0/src/libs/3rdparty/yaml-cpp/src/emitterutils.cpp.omv~	2025-01-01 00:00:00.000000000 +0000
+++ qt-creator-opensource-src-19.0.0/src/libs/3rdparty/yaml-cpp/src/emitterutils.cpp	2026-07-26 00:00:00.000000000 +0000
@@ -1,5 +1,6 @@
 #include <algorithm>
 #include <iomanip>
+#include <cstdint>
 #include <sstream>
 
 #include "emitterutils.h"
