--- qt-creator-opensource-src-19.0.0.orig/src/plugins/clangformat/clangformatutils.cpp	2026-03-10 10:51:20.000000000 +0100
+++ qt-creator-opensource-src-19.0.0/src/plugins/clangformat/clangformatutils.cpp	2026-07-26 18:37:23.024060957 +0200
@@ -66,7 +66,11 @@ clang::format::FormatStyle calculateQtcS
     style.AllowAllParametersOfDeclarationOnNextLine = true;
     style.AllowShortBlocksOnASingleLine = FormatStyle::SBS_Never;
     style.AllowShortCaseLabelsOnASingleLine = false;
+#if LLVM_VERSION_MAJOR >= 23
+    style.AllowShortFunctionsOnASingleLine = {false, true, false};
+#else
     style.AllowShortFunctionsOnASingleLine = FormatStyle::SFS_Inline;
+#endif
     style.AllowShortIfStatementsOnASingleLine = FormatStyle::SIS_Never;
     style.AllowShortLoopsOnASingleLine = false;
     style.AlwaysBreakBeforeMultilineStrings = false;
@@ -77,8 +81,14 @@ clang::format::FormatStyle calculateQtcS
     style.AlwaysBreakAfterReturnType = FormatStyle::RTBS_None;
     style.AlwaysBreakTemplateDeclarations = FormatStyle::BTDS_Yes;
 #endif
+#if LLVM_VERSION_MAJOR >= 23
+    style.PackArguments = { FormatStyle::BPAS_OnePerLine };
+#else
     style.BinPackArguments = false;
-#if LLVM_VERSION_MAJOR >= 20
+#endif
+#if LLVM_VERSION_MAJOR >= 23
+    style.PackParameters = { FormatStyle::BPPS_OnePerLine };
+#elif LLVM_VERSION_MAJOR >= 20
     style.BinPackParameters = FormatStyle::BPPS_OnePerLine;
 #else
     style.BinPackParameters = false;
