diff -up qt-creator-opensource-src-14.0.2/src/plugins/clangformat/clangformatutils.cpp.omv~ qt-creator-opensource-src-14.0.2/src/plugins/clangformat/clangformatutils.cpp
--- qt-creator-opensource-src-14.0.2/src/plugins/clangformat/clangformatutils.cpp.omv~	2024-10-03 14:28:27.559911617 +0200
+++ qt-creator-opensource-src-14.0.2/src/plugins/clangformat/clangformatutils.cpp	2024-10-03 16:36:06.810636184 +0200
@@ -117,7 +117,7 @@ clang::format::FormatStyle calculateQtcS
     style.IndentWrappedFunctionNames = false;
     style.JavaScriptQuotes = FormatStyle::JSQS_Leave;
     style.JavaScriptWrapImports = true;
-    style.KeepEmptyLinesAtTheStartOfBlocks = false;
+    style.KeepEmptyLines = {false, false, false};
     // Do not add QT_BEGIN_NAMESPACE/QT_END_NAMESPACE as this will indent lines in between.
     style.MacroBlockBegin = "";
     style.MacroBlockEnd = "";
