--- qt-creator-2.8.0-rc-src/src/libs/ssh/ssh.pro.orig	2013-06-26 18:35:04.000000000 +0200
+++ qt-creator-2.8.0-rc-src/src/libs/ssh/ssh.pro	2013-07-04 14:54:14.366257550 +0200
@@ -67,3 +67,5 @@ HEADERS = $$PWD/sshsendfacility_p.h \
 FORMS = $$PWD/sshkeycreationdialog.ui
 
 include(../3rdparty/botan/botan.pri)
+
+unix: LIBS += -ldl
--- qt-creator-2.8.0-rc-src/src/libs/utils/utils.pro.orig	2013-07-04 14:54:14.366257550 +0200
+++ qt-creator-2.8.0-rc-src/src/libs/utils/utils.pro	2013-07-04 14:54:40.306545708 +0200
@@ -20,3 +20,4 @@ lessThan(QT_MAJOR_VERSION, 5) {
 win32: LIBS += -luser32
 # PortsGatherer
 win32: LIBS += -liphlpapi -lws2_32
+unix: LIBS += -lX11
