diff -up qt-creator-opensource-src-13.0.0-beta2/src/plugins/clangformat/CMakeLists.txt.omv~ qt-creator-opensource-src-13.0.0-beta2/src/plugins/clangformat/CMakeLists.txt
--- qt-creator-opensource-src-13.0.0-beta2/src/plugins/clangformat/CMakeLists.txt.omv~	2024-02-29 15:25:56.204097655 +0100
+++ qt-creator-opensource-src-13.0.0-beta2/src/plugins/clangformat/CMakeLists.txt	2024-02-29 15:44:06.948355399 +0100
@@ -26,7 +26,7 @@ endif()
 
 extend_qtc_plugin(ClangFormat
   CONDITION UNIX AND NOT APPLE
-  PROPERTIES LINK_FLAGS "-Wl,--exclude-libs,ALL"
+  PROPERTIES LINK_FLAGS "-Wl,--exclude-libs,ALL -lLLVM"
 )
 
 extend_qtc_plugin(ClangFormat
diff -up qt-creator-opensource-src-13.0.0-beta2/src/plugins/projectexplorer/runcontrol.cpp.omv~ qt-creator-opensource-src-13.0.0-beta2/src/plugins/projectexplorer/runcontrol.cpp
--- qt-creator-opensource-src-13.0.0-beta2/src/plugins/projectexplorer/runcontrol.cpp.omv~	2024-02-29 15:02:36.176274142 +0100
+++ qt-creator-opensource-src-13.0.0-beta2/src/plugins/projectexplorer/runcontrol.cpp	2024-02-29 15:04:22.597573679 +0100
@@ -418,7 +418,7 @@ void RunControl::setDevice(const IDevice
     QTC_CHECK(!d->device);
     d->device = device;
 #ifdef WITH_JOURNALD
-    if (!device.isNull() && device->type() == ProjectExplorer::Constants::DESKTOP_DEVICE_TYPE) {
+    if (device && device->type() == ProjectExplorer::Constants::DESKTOP_DEVICE_TYPE) {
         JournaldWatcher::instance()->subscribe(this, [this](const JournaldWatcher::LogEntry &entry) {
 
             if (entry.value("_MACHINE_ID") != JournaldWatcher::instance()->machineId())
